examples/example-protocol/src/main.rs

```diff
@@ -220,6 +220,8 @@ fp_export! {
 const VERSION: &str = "1.0.0";
 const AUTHORS: &str = r#"["Fiberplane <info@fiberplane.com>"]"#;
 const NAME: &str = "example-bindings";
+const DESCRIPTION: &str = r#"Bindings to the fp-bindgen example protocol"#;
+const LICENSE: &str = r#"MIT OR Apache-2.0"#;
```

Comment:
I don't think these need to be raw strings.

Reply:
> Looks good, but I think we should indeed take the opportunity to start using builders for the binding config types, so that we won't need to do breaking changes if we add more fields.
> 
> I'm also slightly wary of the `maybe_quote_str` approach. It works, but it's not very discoverable. Why does it work for `version`, but not for `authors`? They use the same types, so I would expect the same behavior. I think we should either document this clearly, or make it explicit in the type system.

Authors could use the same processing, indeed. How do you feel about introducing the enum to indicate whether it should be a string or a workspace ref? That would everything much more explicit.

I'll rework this to use a builder regardless.
 
> Did you try your changes with the provider bindings yet?

Yes, that works fine. I did notice that `fp-bindgen-support` got patched to refer to the `workspace` version instead of the actual version. We don't support that yet, but we could add that in a seperate PR I think (maybe even using the same new enum).

Reply:
> How do you feel about introducing the enum to indicate whether it should be a string or a workspace ref?

Sounds good!

> Yes, that works fine. I did notice that the dependency for fp-bindgen-support got patched in the generated Cargo.toml to refer to the workspace version instead of the actual version. We don't support that yet, but we could add that in a separate PR I think (maybe even using the same new enum).

Huh, I thought I had fixed that in the protocol already, but maybe I'm mistaken. Wasn't that just a case of explicitly specifying `CargoDependency::from_workspace()` in the `main.rs`?

Reply:
Right now `fp-bindgen-support` always gets added with the same version as `fp-bindgen` itself, in the generator.

We could add some configuration to also allow the workspace version to be used, in the future.

Reply:
Yeah, it does, but dependencies get merged, allowing users to overwrite the version set in the generator. It's what we do here: https://github.com/fiberplane/fiberplane/blob/main/fiberplane-provider-protocol/src/main.rs#L87